lib/elixir/lib/access.ex

```diff
@@ -215,7 +219,7 @@ defmodule Access do
 
   See the implementations for `Map.pop/3` or `Keyword.pop/3` for more examples.
   """
-  @callback pop(term :: t, key) :: {value, t}
+  @callback pop(container, key) :: {value, container} when container: t
```

Comment:
true. the fact is that the `container` type thing came at the very end of my work, as I run Dialyzer and I run into a few issues. I will work on it